### PR TITLE
[2.3.27] added matching fields to subformat 4

### PR DESCRIPTION
**Pull request by mzieg**

### CODE REVIEW

File: WasatchNET/EEPROM.cs [hunk 1]
```diff
@@ -881,6 +881,39 @@ public byte startupScansToAverage
...
+        public byte matchingMinRampPixels
+        {
+            get { return _matchingMinRampPixels; }            
+            set
+            {
+                _matchingMinRampPixels = value;
+                EEPROMChanged?.Invoke(this, new EventArgs());
+            }
+        }
+        byte _matchingMinRampPixels = 10;
```
Comment (trestoh) [inline WasatchNET/EEPROM.cs:893]: I'm curious at what this is meant to do. I should probably look at the previous PR...

File: WasatchNET/EEPROM.cs [hunk 2]
```diff
@@ -1879,15 +1915,12 @@ protected bool writeParse()
...
-                    {
-                        logger.error("EEPROM.writeLibrary inapplicable on subformat 0x{0:2x}", subformat);
-                        return false;
-                    }
+                    if (!ParseData.writeByte(libraryType,               pages[7], 0)) return false;
+                    if (!ParseData.writeUInt16(libraryID,               pages[7], 1)) return false;
+                    if (!ParseData.writeByte(startupScansToAverage,     pages[7], 3)) return false;
+                    if (!ParseData.writeByte(matchingMinRampPixels,     pages[7], 4)) return false;
+                    if (!ParseData.writeUInt16(matchingMinPeakHeight,   pages[7], 5)) return false;
+                    if (!ParseData.writeByte(matchingThreshold,         pages[7], 7)) return false;
```
Comment (trestoh) [inline WasatchNET/EEPROM.cs:1923]: I need to review the library features. These presuppose a large library of EEPROM pages though, correct?

**Review by trestoh — APPROVED** [review]
I need to review some of these features a bit, but they look to be implemented properly

File: WasatchNET/EEPROM.cs (hunk 1 above)
Comment (mzieg) [reply WasatchNET/EEPROM.cs:893]: Technically it's just a named storage field, that doesn't "do" anything in the driver.  However, it does do something in the spectrometer (at least some models :-)  Can discuss more offline.

File: WasatchNET/EEPROM.cs (hunk 2 above)
Comment (mzieg) [reply WasatchNET/EEPROM.cs:1923]: The "library" is essentially a stack of 61 EEPROM pages (zero-indexed 10 through 70), enough to store 61 x 64 = 3904 bytes of data, i.e. 1952 uint16, or the size of one SiG spectrum.  I'll send you a link with additional context.

**Pull request merged**